File: service/app/src/test/java/io/miragon/blueprint/adapter/outbound/notification/LoggingNotificationAdapterTest.java
package io.miragon.blueprint.adapter.outbound.notification;

import static io.miragon.blueprint.domain.leasing.TestObjectBuilder.testLeasingApplication;
import static org.assertj.core.api.Assertions.assertThatCode;

import io.miragon.blueprint.domain.leasing.LeasingApplication;
import org.junit.jupiter.api.Test;

class LoggingNotificationAdapterTest {

    private final LoggingNotificationAdapter underTest = new LoggingNotificationAdapter();

    @Test
    void sendLogsTheNotificationWithoutError() {

        // given: a subject and an application
        LeasingApplication application = testLeasingApplication().build();

        // when / then: sending the notification completes without throwing
        assertThatCode(() -> underTest.send("Please review and sign your leasing contract", application))
                .doesNotThrowAnyException();
    }
}

File: service/app/src/test/kotlin/io/miragon/blueprint/adapter/outbound/notification/LoggingNotificationAdapterTest.kt
package io.miragon.blueprint.adapter.outbound.notification

import io.miragon.blueprint.domain.leasing.testLeasingApplication
import org.assertj.core.api.Assertions.assertThatCode
import org.junit.jupiter.api.Test

class LoggingNotificationAdapterTest {

    private val underTest = LoggingNotificationAdapter()

    @Test
    fun `send logs the notification without error`() {

        // given: a subject and an application
        val application = testLeasingApplication()

        // when / then: sending the notification completes without throwing
        assertThatCode { underTest.send("Please review and sign your leasing contract", application) }
            .doesNotThrowAnyException()
    }
}
